deps/ncrypto/ncrypto.cc

```diff
@@ -4191,6 +4191,28 @@ DataPointer hashDigest(const Buffer<const unsigned char>& buf,
   return data.resize(result_size);
 }
 
+DataPointer xofHashDigest(const Buffer<const unsigned char>& buf,
+                          const EVP_MD* md,
+                          size_t output_length) {
+  if (md == nullptr) return {};
+
+  if (output_length == 0) {
+    auto data = DataPointer::Alloc(1);
+    if (!data) return {};
+    return data.resize(0);
```

Comment:
@jasnell do you have a different suggestion on how to alloc a 0-length DataPointer to return here?

```
auto data = DataPointer::Alloc(1);
if (!data) return {};
return data.resize(0);
```

Reply:
I think when the length is 0 it would be fine to just special case from JS land and don't go into C++ at all. If outputEncoding is 'buffer' return an empty buffer from JS land. Otherwise return an empty string.

Reply:
That's what was there before but then this test wasn't passing.

https://github.com/nodejs/node/pull/58121/files#diff-afbb57b034233cb3104065ae659d3702a9c2ad9cc8a4017e01baebcd551d5413R120-R121 (non-XOF with invalid (0) outputLength)